tests/autoyast/prepare_profile.pm

```diff
@@ -31,6 +32,23 @@ sub run {
     # Get file from data directory
     my $profile = get_test_data($path);
 
+    # Add registration block for cloned autoyast profiles due to
+    # the registration code is never clonned as it is not stored in the system anywhere
+    if (get_var('AUTOYAST_ADD_REGISTRATION')) {
+        # Create registration block
+        my $suse_register = << "EOF";
+<suse_register>
+    <do_registration config:type="boolean">true</do_registration>
+    <email/>
+    <reg_code>{{SCC_REGCODE}}</reg_code>
+    <install_updates config:type="boolean">true</install_updates>
+    <reg_server>{{SCC_URL}}</reg_server>
+  </suse_register>
+EOF
+        # Inject registration block
+        $profile =~ s/(?<add_on><add-on>)/$suse_register  $+{add_on}/g;
```

Comment:
So what I don't like about this, is that now <add-on> section has to be in the profile. So, I wonder why we don't want to just inject this into `<profile>`? Then we could have used this for all the profiles.

Reply:
We might have `add-on` in all of our possible cases (registration without addons?) but it could be somehow. Addressed.